src/reducers/signupReducer.js

```diff
@@ -0,0 +1,22 @@
+// @call type going to be used..
+import { SET_SIGNUP_FORM_INPUT, USER_ERRORS } from '../actions/type';
+
+const initialState = {};
+const reducer = (state = initialState, { type, payload }) => {
+  switch (type) {
+    case SET_SIGNUP_FORM_INPUT:
+      return {
+        ...state,
```

Comment:
Why are you spreading an empty state? Your initial state shoust have someting like
``` 
{
user: {
*whatever data is retured when you register a user. which is empty initially* 
},
errors: null
}
```